repository/user_id_provider.go

```diff
@@ -6,21 +6,20 @@ import (
 	"github.com/casdoor/casdoor/object"
 )
 
-func (r *Repo) GetUserIdProvider(ctx context.Context, owner, providerName, usernameFromIdp string) (*object.UserIdProvider, error) {
-	if owner == "" || providerName == "" || usernameFromIdp == "" {
+func (r *Repo) GetUserIdProvider(ctx context.Context, userIdProvider *object.UserIdProvider) (*object.UserIdProvider, error) {
+	if userIdProvider.Owner == "" || ((userIdProvider.ProviderName == "") == (userIdProvider.LdapId == "")) || userIdProvider.UsernameFromIdp == "" {
```

Comment:
I think it needs to be simplified

Reply:
Couldn't find a better solution. https://github.com/casgate/casgate/pull/146#discussion_r1579502078